### PR TITLE
fix query option `append`

### PR DESCRIPTION
**Pull request by xithan**
#process_names was called with wrong number of arguments

### CODE REVIEW

File: card/spec/lib/card/query_spec.rb [hunk 1]
```diff
@@ -24,6 +24,11 @@
       is_expected.to eq(%w(C+A D+A F+A))
     end
 
+    it "appends names to card results" do
+      res = Card::Query.run name: [:in, "C", "D", "F", "Z"], append: "A"
+      expect(res).to eq([Card["C+A"],Card["D+A"], Card["F+A"], nil])
+    end
```
Comment (xithan) [inline card/spec/lib/card/query_spec.rb]: Our query specs are insufficient. We only test name results.

File: card/lib/card/query/run.rb [hunk 2]
```diff
@@ -72,7 +76,7 @@ def alter_results?
 
       def card_result record, _field
         if alter_results?
-          Card.fetch process_name(record["name"])
+          Card.fetch alter_result(record["name"])
         else
           fetch_or_instantiate record
```
Comment (xithan) [inline card/lib/card/query/run.rb]: Is there a reason that we don't cache when we alter the result?

Comment (ethn) [reply card/lib/card/query/run.rb]: #fetch does cache.  but we can't instantiate an altered result.  (all the information needed for normal instantiation is in the record, but the record isn't sufficient for an altered result

File: card/spec/lib/card/query_spec.rb (hunk 1 above)
Comment (xithan) [reply card/spec/lib/card/query_spec.rb]: What should a search with an `append` statement return if the card doesn't exist? A new card or nothing? I think returning `nil`, what it currently does, will cause trouble somewhere someday.

Comment (ethn) [reply card/spec/lib/card/query_spec.rb]: There should be a `new: {}` there.  I broke it in this commit: https://github.com/xithan/decko/commit/6f1c02a8f7787ab0641385b5d398bd8a49ab86c4#diff-6b22a1f8bd11baaec00638be4a7d3814  (which was merged fairly recently)

Comment (ethn) [reply card/spec/lib/card/query_spec.rb]: (only needed in card_result.  `Card.fetch alter_result(record["name"]), new: {}`

**Pull request merged**